gnome-image-installer/pages/finished/gis-finished-page.c

```diff
@@ -134,42 +134,54 @@ toggle_leds (GisPage *page)
 #define EOS_IMAGE_VERSION_ALT_PATH "/"
 
 static gchar *
-get_image_version (const gchar *path)
+gis_page_util_get_image_version (const gchar *path,
+                                 GError     **error)
 {
   ssize_t attrsize;
-  gchar *value;
+  g_autofree gchar *value = NULL;
+  int errsv;
 
   g_return_val_if_fail (path != NULL, NULL);
 
   attrsize = getxattr (path, EOS_IMAGE_VERSION_XATTR, NULL, 0);
-  if (attrsize < 0) {
-    g_message ("Error examining " EOS_IMAGE_VERSION_XATTR " on %s: %s",
-               path, g_strerror (errno));
-    return NULL;
-  }
+  if (attrsize < 0)
+    goto error;
 
   value = g_malloc (attrsize + 1);
   value[attrsize] = 0;
 
   attrsize = getxattr (path, EOS_IMAGE_VERSION_XATTR, value,
                        attrsize);
-  if (attrsize < 0) {
-    g_warning ("Error reading " EOS_IMAGE_VERSION_XATTR " on %s: %s",
-               path, g_strerror (errno));
-    g_free (value);
-    return NULL;
-  }
-
-  return value;
+  if (attrsize < 0)
+    goto error;
+
+  return g_steal_pointer (&value);
+
+error:
+  errsv = errno;
+  g_set_error (error, G_IO_ERROR, g_io_error_from_errno (errsv),
+               "Error examining " EOS_IMAGE_VERSION_XATTR " on %s: %s",
+               path, g_strerror (errsv));
+  return NULL;
 }
 
 static gboolean
 is_eosdvd (void)
 {
-  g_autofree gchar *image_version = get_image_version (EOS_IMAGE_VERSION_PATH);
+  g_autoptr(GError) error1 = NULL;
+  g_autoptr(GError) error2 = NULL;
+  g_autofree char *image_version =
+    gis_page_util_get_image_version (EOS_IMAGE_VERSION_PATH, &error1);
+
+  if (image_version == NULL)
+    image_version =
+      gis_page_util_get_image_version (EOS_IMAGE_VERSION_ALT_PATH, &error2);
 
   if (image_version == NULL)
-    image_version = get_image_version (EOS_IMAGE_VERSION_ALT_PATH);
+    {
+      g_warning ("%s", error1->message);
+      g_warning ("%s", error2->message);
+    }
```

Comment:
What errors can be returned here? Some more context would be nice, eg `g_warning ("Error reading standard image version path: %s")`

Also, should we show a warning about being able to read from the standard image version path if we were able to successfully read the alternative one?

Reply:
> What errors can be returned here? Some more context would be nice, eg g_warning ("Error reading standard image version path: %s")

There is only one error, which you can see by scrolling up to the function directly above this one:

```c
  g_set_error (error, G_IO_ERROR, g_io_error_from_errno (errsv),
               "Error examining " EOS_IMAGE_VERSION_XATTR " on %s: %s",
               path, g_strerror (errsv));
```

It includes all of the context you describe.

> Also, should we show a warning about being able to read from the standard image version path if we were able to successfully read the alternative one?

The whole point of this change is not to do that! Here is the commit message:

> page-util: remove spurious message about reading eos-image-version
> 
> The eos-image-version xattr is stored on the real root directory of the
> root filesystem. On converted systems, this is `/sysroot`, while on
> unconverted systems, this is `/`. We check them in that order.
> 
> Previously, when run on a converted system, we would always print a
> message about being unable to read this attribute from `/sysroot`. But
> this is expected! It's only a problem if we can't read it from `/`
> either; so, only print a message in that case.

Reply:
Makes sense (looking at the error above)